demo/site/src/middleware/predefinedPages.ts

```diff
@@ -49,7 +48,15 @@ const predefinedPagesQuery = gql`
     }
 `;
 
-const graphQLFetch = createGraphQLFetch();
+const graphQLFetch = createGraphQLFetchLibrary(
```

Comment:
I think that should be somewhere re-usable (though naming will be difficult)

Or do you think there are no other use-cases for accessing the api in a middleware?